### PR TITLE
Bump up to python 3.7 for Azure Pipelines

### PR DESCRIPTION
**Pull request by youknowone**
Split from #1703

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -8,8 +8,8 @@ jobs:
     vmImage: 'vs2017-win2016'
   strategy:
     matrix:
-      Python36:
-        python.version: '3.6'
+      Python38:
+        python.version: '3.8'
```
Comment (coolreader18) [inline azure-pipelines.yml]: I think we can embed this directly in the `UsePythonVersion` task; I'm pretty sure this is what's causing the 2 different Azure jobs to run.

**Comment by youknowone** [comment]
windows snippets test fail with python 3.8 but I don't have windows macine.

**Comment by palaviv** [comment]
It seems like [os.path.realpath](https://docs.python.org/3/library/os.path.html#os.path.realpath) was changed in 3.8.

**Comment by coolreader18** [comment]
The os.path.realpath implementation for 3.8 is [pretty complex](https://github.com/python/cpython/blob/3.8/Lib/ntpath.py#L537) and requires a fairly complex [`os` module function](https://github.com/python/cpython/blob/3.8/Modules/posixmodule.c#L3880), it might be best to comment it out for windows right now and put a TODO.

**Comment by youknowone** [comment]
Changed to python3.7 for azure

**Pull request merged**